File: scripts/verify_migration.py
import mysql.connector
import json
import os

def load_config():
    # Try to load from Line-bot-llm-mysql/src/config.ts (parsing TS file is hard, let's try to find a json config or use env vars if available)
    # Or better, use the config from ebook/config.json if it has DB creds?
    # ebook/config.json usually has API URL.
    # Let's try to read from .env file in Line-bot-llm-mysql if it exists.
    
    env_path = os.path.join('Line-bot-llm-mysql', '.env')
    config = {}
    if os.path.exists(env_path):
        with open(env_path, 'r') as f:
            for line in f:
                if '=' in line:
                    key, value = line.strip().split('=', 1)
                    config[key] = value
    return config

def verify_migration():
    config = load_config()
    
    # Fallback to default credentials if not found in .env (assuming local dev defaults)
    host = config.get('DB_HOST', 'localhost')
    user = config.get('DB_USER', 'root')
    password = config.get('DB_PASSWORD', 'password') # Replace with actual default if known, or try empty
    database = config.get('DB_NAME', 'library_db')
    
    print(f"Connecting to {host} as {user}...")
    
    try:
        conn = mysql.connector.connect(
            host=host,
            user=user,
            password=password,
            database=database
        )
        cursor = conn.cursor()
        
        # Check dharma_books table
        cursor.execute("SHOW TABLES LIKE 'dharma_books'")
        result = cursor.fetchone()
        if result:
            print("SUCCESS: Table 'dharma_books' exists.")
        else:
            print("FAILURE: Table 'dharma_books' does NOT exist.")
            
        # Check subscribers table for subscribed_videos column
        cursor.execute("SHOW COLUMNS FROM subscribers LIKE 'subscribed_videos'")
        result = cursor.fetchone()
        if result:
            print("SUCCESS: Column 'subscribed_videos' exists in 'subscribers'.")
        else:
            print("FAILURE: Column 'subscribed_videos' does NOT exist in 'subscribers'.")
            
        cursor.close()
        conn.close()
        
    except mysql.connector.Error as err:
        print(f"Error: {err}")

if __name__ == "__main__":
    verify_migration()

File: scripts/verify_migration.js
const mysql = require('mysql2/promise');
const fs = require('fs');
const path = require('path');

// Load config from environment or hardcoded defaults (matching the project)
// In a real scenario we would parse .env or config.ts, but for this script we'll try to be robust
const config = {
    host: process.env.DB_HOST || 'localhost',
    user: process.env.DB_USER || 'root',
    password: process.env.DB_PASSWORD || 'password',
    database: process.env.DB_NAME || 'library_db'
};

async function verify() {
    console.log(`Connecting to ${config.host} as ${config.user}...`);
    let connection;
    try {
        connection = await mysql.createConnection(config);

        // Check dharma_books table
        const [tables] = await connection.execute("SHOW TABLES LIKE 'dharma_books'");
        if (tables.length > 0) {
            console.log("SUCCESS: Table 'dharma_books' exists.");
        } else {
            console.log("FAILURE: Table 'dharma_books' does NOT exist.");

            // Try to run the migration SQL directly if it fails
            console.log("Attempting to run migration SQL...");
            const sqlPath = path.join(__dirname, '../Line-bot-llm-mysql/migrations/002_add_dharma_books_and_videos.sql');
            if (fs.existsSync(sqlPath)) {
                const sql = fs.readFileSync(sqlPath, 'utf8');
                const statements = sql.split(';').filter(s => s.trim());
                for (const stmt of statements) {
                    if (stmt.trim()) {
                        try {
                            await connection.execute(stmt);
                            console.log("Executed SQL statement.");
                        } catch (err) {
                            console.log(`Error executing statement: ${err.message}`);
                        }
                    }
                }
                console.log("Migration SQL execution completed.");
            } else {
                console.log(`Migration file not found at ${sqlPath}`);
            }
        }

        // Check subscribers table
        const [columns] = await connection.execute("SHOW COLUMNS FROM subscribers LIKE 'subscribed_videos'");
        if (columns.length > 0) {
            console.log("SUCCESS: Column 'subscribed_videos' exists in 'subscribers'.");
        } else {
            console.log("FAILURE: Column 'subscribed_videos' does NOT exist in 'subscribers'.");
        }

    } catch (err) {
        console.error("Error:", err.message);
    } finally {
        if (connection) await connection.end();
    }
}

verify();
